lib/location/location_core.c

```diff
@@ -506,19 +542,48 @@ static void location_core_periodic_work_fn(struct k_work *work)
 	location_core_location_get_pos(&current_config);
 }
 
-static void location_core_timeout_work_fn(struct k_work *work)
+static void location_core_method_timeout_work_fn(struct k_work *work)
 {
 	enum location_method current_method =
 		current_config.methods[current_method_index].method;
 
 	ARG_UNUSED(work);
 
-	LOG_WRN("Timeout occurred");
+	LOG_WRN("Method specific timeout expired");
 
 	location_method_api_get(current_method)->cancel();
 	location_core_event_cb_timeout();
 }
 
+static void location_core_timeout_work_fn(struct k_work *work)
+{
+	enum location_method current_method =
+		current_config.methods[current_method_index].method;
+
+	ARG_UNUSED(work);
+
+	LOG_WRN("Timeout for entire location request expired");
+
+	location_method_api_get(current_method)->cancel();
+	/* config->timeout needs to expire without fallbacks */
+
+	current_event_data.id = LOCATION_EVT_TIMEOUT;
+	event_handler(&current_event_data);
+
+	k_work_cancel_delayable(&location_core_timeout_work);
+
+	if (current_config.interval > 0) {
+		k_work_schedule_for_queue(
+			location_core_work_queue_get(),
+			&location_periodic_work,
+			K_SECONDS(current_config.interval));
+	} else {
+		location_core_current_config_clear();
+
+		k_sem_give(&location_core_sem);
```

Comment:
This code is probably somewhere else as well, if so, it could be moved to a new function with a descriptive name.

Reply:
Closing as this is same copy-paste issue as previous comment https://github.com/nrfconnect/sdk-nrf/pull/8631#discussion_r993122623